### PR TITLE
Add AsSemigroup method for strong semilattices of groups

### PR DESCRIPTION
**Pull request by flsmith**
This PR adds a method for constructing Clifford semigroups as partial permutation semigroups, given a semilattice, groups, and homomorphisms.

You can choose between an easily-computed representation (the default) and a more difficult but possibly smaller degree representation. The smaller degree representation is based on 

"The minimal faithful degree of a semilattice of groups"
David Easdown
J. Austral. Math. Soc. (Series A) 45 (1988), 341-350

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
@flsmith and I just discussed this in person, and @flsmith is going to check if there is overlap between the code in this PR and `SmallerDegreePartialPermRepresentation`.

**Comment by wilfwilson** [comment]
What could be the possible outcome of this checking? That this PR somehow gets revised accordingly?

**Comment by james-d-mitchell** [comment]
Yes, that's right, it isn't clear if there is overlap with what's already in Semigroups, with respect to `SmallerDegreePartialPermRepresentation` so this should be figured out, and if there's an overlap it should probably be removed.

**Comment by flsmith** [comment]
I've now removed the overlap, so the fast method which does not attempt to minimise the degree is used. You can then use `SmallerDegreePartialPermRepresentation` on the result.

File: doc/oper.xml [hunk 1]
```diff
@@ -1589,3 +1589,45 @@ gap> Size(M);
     </Description>
   </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="AsCliffordSemigroup">
```
Comment (ChristopherRussell) [inline doc/oper.xml]: Should you change this to <#GAPDoc Label="AsSemigroup"> ?

File: doc/z-chap4.xml [hunk 2]
```diff
@@ -70,6 +70,7 @@
 
   <Section><Heading>Associated semigroups</Heading>
     <#Include Label="AsSemigroup">
+    <#Include Label="AsCliffordSemigroup">
```
Comment (ChristopherRussell) [inline doc/z-chap4.xml]: Should this also be "AsSemigroup"?

**Comment by flsmith** [comment]
Resolved comments, ready for merge

**Comment by james-d-mitchell** [comment]
Can you please rebase this onto the current `master` branch? @flsmith, thanks!

**Pull request merged**